docs/writing-docs/tutorials/basics.md

```diff
@@ -267,6 +267,33 @@ Gray blocks will be generated when not all datatypes are known (e.g. `mySprite`
 If you don't see an error message, try adding the `dbg=1` URL parameter and reload. This will output some information about each blockconfig to the console, and should provide a clue about what is failing.
 
 
+## Code Validation (`validation.local` and `validation.global` sections)
+
+If you want to enable code validation in your tutorial, you can do so by adding a validation section. As with [`blockconfig` above](#reconfiguring-blocks-in-the-toolbox-blockconfiglocal-and-blockconfigglobal-sections), there are two types of sections: `validation.global` and `validation.local`. Global can be anywhere in the markdown and applies to all steps in the tutorial. Local must appear within a specific step, and applies only to that step. Local takes precedence over global.
+
+Within a validation section, you may specify which validators you want to enable and properties for those validators using the same syntax we use for writing Skillmaps (see [skillmap structure](..\skillmaps.md#skillmap-structure)).
+
+### Validators
+
+Currently, only one validator exists: the `BlocksExistValidator`. This validator looks at highlighted blocks in the answer key and confirms that, for each highlighted block, the user's code contains at least one block of the same type. It does *not* validate the parameters passed into the block.
```

Comment:
I couldn't find any docs on highlights in the answer key. Do we have those? If so, I'd like to link to them here.

Reply:
I don't know if we do.  We should probably add //@highlight (and if it does exist, update to explain that you can now highlight multiple blocks). 

Related, do we have documentation on @hide and @collapse?

Reply:
looks like highlight is mentioned here https://makecode.com/writing-docs/snippets#highlight, I should add hide /collapse there as well

Reply:
Nice, thanks! I've added a link to that section.